rq/registry.py

```diff
@@ -334,6 +333,66 @@ def cleanup(self):
         pass
 
 
+class QueuedJobRegistry(BaseRegistry):
+    key_template = 'rq:queued:{0}'
+
+    def get_expired_job_ids(self, timestamp=None):
+        raise NotImplementedError
+
+    def cleanup(self):
+        """This method is only here to prevent errors because this method is
+        automatically called by `count()` and `get_job_ids()` methods
+        implemented in BaseRegistry."""
+        pass
+
+    def requeue_stuck_jobs(self):
+        """This method requeues jobs that are popped off the queue but still belong to QueuedJobRegistry
+        It is not defined in cleanup since we don't want this being called everytime count or get_job_ids is called
+        """
+        job_ids = self.get_job_ids()
+        for job_id in job_ids:
```

Comment:
This could be expensive if we have lots of jobs in `QueuedJobRegistry` and sometimes people have millions of jobs enqueued. I wonder if we have a way to only check a subset of the jobs.